### PR TITLE
refactor: list and list item implementation

### PR DESCRIPTION
**Pull request by angusfretwell**
@ehyland @dougmcclurg Ran into some trouble trying to use this component. I assume you're using it already so would be good to get an idea of how you're using it...

There are a few changes here:
- Switched the `list-style-position` to `inside` and removed left padding. This means that the list lines up with any text above/below it. Previously the padding was such that it *almost* lined up, but off by 1-2px which looked odd. The only downside here is that you lose indentation when it wraps. 
- Added a gutter to the list when there are multiple columns. Previously when it was `list-style-position` outside and without any gutter, the text and bullets could overlap. 
- Refactored the implementation to use `responsiveStyle` from `styled-system` which is a little easier to read IMO. This will rely on `columns` being null by default to support margins collapsing (i.e. not setting `flex`); I think this is OK since setting `<List columns={1} />` seems unlikely.

### CODE REVIEW

**Comment by allanpope** [comment]
@angusfretwell I think its a design question but it would be nice to keep the indentation on list items when text wraps

File: packages/components/src/List/List.story.js [hunk 1]
```diff
@@ -22,31 +23,24 @@ const flatList = props => (
...
-  }))
+  .add('default', () =>
+    flatList({
+      ordered: boolean('Ordered', false),
+      flush: boolean('Flush', false),
+    }))
+  .add('columns', () =>
+    flatList({
+      ordered: boolean('Ordered', false),
+      columns: array('Columns', [1, 2, 3]),
```
Comment (angusfretwell) [inline packages/components/src/List/List.story.js]: @ehyland This worked for me? 🤷‍♂️

File: packages/components/src/List/List.js [hunk 2]
```diff
@@ -1,61 +1,55 @@
...
-    ...responsiveColumns.map((numColumns, index) => css`
-      @media (min-width: ${themeGet(`breakpoints.${index}`)}) {
-        ${ListItem} {
-          flex: 1 1 ${100 / numColumns}%;
-        }
-      }
-    `),
-  ];
-};
+const columns = responsiveStyle({
```
Comment (ehyland) [inline packages/components/src/List/List.js]: Damn, so much cleaner

Comment (angusfretwell) [reply packages/components/src/List/List.js]: 😄

**Pull request merged**